course-v2/layouts/partials/resources_header.html

```diff
@@ -1,22 +1,37 @@
-{{ $courseData := site.Data.course }}
-{{ $zenDeskUrl := "https://mitocw.zendesk.com/hc/en-us/articles/4414681093659-I-have-downloaded-an-MIT-OpenCourseWare-course-but-I-can-t-access-the-materials-How-do-I-get-started-" }}
-{{ $trimmedBaseUrl := strings.TrimSuffix "/" site.BaseURL }}
-{{ $shortId := printf "%v-%v-%v" $courseData.primary_course_number (lower $courseData.term) $courseData.year }}
-{{ $downloadCourseUrl := printf "%v/%v.zip" $trimmedBaseUrl $shortId }}
+{{- $courseData := site.Data.course -}}
+{{- $zenDeskUrl := "https://mitocw.zendesk.com/hc/en-us/articles/4414681093659-I-have-downloaded-an-MIT-OpenCourseWare-course-but-I-can-t-access-the-materials-How-do-I-get-started-" -}}
+{{- $trimmedBaseUrl := strings.TrimSuffix "/" site.BaseURL -}}
+{{- $shortId := printf "%v-%v-%v" $courseData.primary_course_number (lower $courseData.term) $courseData.year -}}
+{{- $downloadCourseUrl := printf "%v/%v.zip" $trimmedBaseUrl $shortId -}}
+{{- $headResponse := resources.GetRemote $downloadCourseUrl (dict "method" "head") -}}
+<!-- 
+  It may seem unintuitive to look explicitly for "unexpected EOF" here. Unfortunately while 
+  Hugo supports setting "method" "head" to perform a HEAD request, it doesn't support checking 
+  the response headers. If we just perform a normal GET here the entire ZIP file will be 
+  downloaded during the build which could vastly balloon the build time or make the build fail.
+
+  A 404 on the OCW site returns an empty body with a 404 status code, which Hugo properly 
+  interprets as nil. In this case the "unexpected EOF" means "I found a file and got a 200 
+  in the response, but the file was empty."
+  
+  The file is empty because we only made a HEAD request.
+-->
+{{- if in $headResponse "unexpected EOF" -}}
```

Comment:
Not sure if this is an optimal approach but I thought about this and couldn't come up with any other robust/better solution.

Reply:
Yea, I'm not super happy with it either as you can see but I think this will work for now. I detailed another potential solution in the issue involving tracking the offline site publish date in `ocw-studio` which is probably more realistically the way to go here but this will prevent the link being shown when the ZIP doesn't exist yet for now.